cmd/agent/app/server.go

```diff
@@ -54,6 +54,11 @@ func NewAgentCommand(a *Agent, o *options.GrpcProxyAgentOptions) *cobra.Command
 }
 
 type Agent struct {
+	connected bool
+}
+
+func (a *Agent) onConnect(status bool) {
```

Comment:
I like the interface that proxy-server defines in `pkg/server/readiness_manager.go`, it seems repeatable for proxy-agent. What do you think?

In the agent case, the ClientSet could expose `cs.Readiness.Ready()` instead of this callback.